### DISCUSSION TITLE
Adopt SAM2 in MLX and enable ANE support

### DISCUSSION THREAD

**Discussion by Silviu-Preoteasa** [ml-explore/mlx-examples#918] [Ideas]
SAM2 by Meta is a very versatile model. The video segmentation especially is extremely useful and the permissive license could drive Mx adoption up. The framerate on Nvidia A6000 and A100 is bad, so I guess this could be a great win if we could optimize for M CPUs and smoke the competition. Could we create a bounty system for this? I'd be happy to support.

I know what we *think* the problem is with ANE and open source projects, but this could be the big opportunity to open access to it and make news.

**Comment by vaibhavoberoi** [comment]
This idea sounds promising and I can see the potential upside to it! Wish to be a part of it!!
